onnxruntime/featurizers_ops/cpu/laglead_operator_transformer.cc

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/common/common.h"
+#include "core/framework/data_types.h"
+#include "core/framework/data_types_internal.h"
+#include "core/framework/op_kernel.h"
+
+#include "Featurizers/LagLeadOperatorFeaturizer.h"
+#include "Featurizers/../Archive.h"
+
+namespace NS = Microsoft::Featurizer;
+
+namespace onnxruntime {
+namespace featurizers {
+
+double ExtractNumFromNullable(float const & nullableValue) {
+  return static_cast<double>(nullableValue);
+}
+
+double ExtractNumFromNullable(double const & nullableValue) {
+  return nullableValue;
+}
+
+template <typename T>
+double ExtractNumFromNullable(nonstd::optional<T> const & nullableValue) {
+  return nullableValue.has_value() ? static_cast<double>(*nullableValue) : NS::Traits<double>::CreateNullValue();
+}
+
+template <typename T>
+struct LagLeadOperatorTransformerImpl {
+  void operator()(OpKernelContext* ctx) const {
+
+    using GrainT = std::vector<std::string>;
+    using EstimatorT = Microsoft::Featurizer::Featurizers::GrainedLagLeadOperatorEstimator<T>;
+    using GrainedInputType = std::tuple<GrainT, T>;
+    using OutputMatrixDataType = typename NS::Traits<T>::nullable_type;
+    using OutputMatrixType = NS::RowMajMatrix<OutputMatrixDataType>;
+    using OutputType = std::tuple<GrainT, OutputMatrixType>;
+
+    //Get the transformer
+    const auto* state_tensor(ctx->Input<Tensor>(0));
+    const uint8_t* const state_data(state_tensor->Data<uint8_t>());
+    Microsoft::Featurizer::Archive archive(state_data, state_tensor->Shape().GetDims()[0]);
+    typename EstimatorT::TransformerType transformer(archive);
+
+    // Get the Grains
+    const auto* grains_tensor(ctx->Input<Tensor>(1));
+    const std::string* grains_data(grains_tensor->Data<std::string>());
+    const int64_t grains_num = grains_tensor->Shape()[1];
+
+    // Get the Target
+    const auto* target_tensor(ctx->Input<Tensor>(2));
+    const T* target_data(target_tensor->Data<T>());
+
+    // Prepare the OutputGrains
+    TensorShape output_grains_shape = grains_tensor->Shape();
+    Tensor* output_grains_tensor(ctx->Output(0, output_grains_shape));
+    std::string* output_grains_data = output_grains_tensor->MutableData<std::string>();
+
+    // Prepare the Output
+    const int64_t output_dim_0 = grains_tensor->Shape()[0];
+
+    double* output_data;
+    bool has_allocate_output_data = false;
+    std::function<void(OutputType)> callback_fn;
```

Comment:
>std::function<void(OutputType)> callback_fn; [](start = 4, length = 44)

This piece of code seems almost totally wrong.
Only floats and doubles are nullable in ORT among the presented datatypes. There are represented as such. Therefore, the output matrix should not require any transformation and should be Mapped/copied directly to the output tensor.

Reply:
fixed